scripts/linux/cross-platform-rust-build.sh

```diff
@@ -0,0 +1,293 @@
+#!/bin/bash
+
+###############################################################################
+# Print usage information pertaining to this script and exit
+###############################################################################
+usage()
+{
+    echo "$SCRIPT_NAME [options]"
+    echo "This script is used for building a few rust components (i.e. broker, edge hub watchdog)."
+    echo "The logic here is similar to the edgelet packaging."
+    echo ""
+    echo "options"
+    echo " --os                      Desired os for build"
+    echo " --arch                    Desired arch for build"
+    echo " --build-path              Desired arch for build"
+    echo " -h, --help                Print this help and exit."
+    exit 1;
+}
+
+###############################################################################
+# Obtain and validate the options supported by this script
+###############################################################################
+process_args()
+{
+    save_next_arg=0
+    for arg in "$@"
+    do
+        if [ $save_next_arg -eq 1 ]; then
+            PACKAGE_OS="$arg"
+            save_next_arg=0
+        elif [ $save_next_arg -eq 2 ]; then
+            PACKAGE_ARCH="$arg"
+            save_next_arg=0
+        elif [ $save_next_arg -eq 3 ]; then
+            BUILD_PATH="$arg"
+            save_next_arg=0
+        else
+            case "$arg" in
+                "--os" ) save_next_arg=1;;
+                "--arch" ) save_next_arg=2;;
+                "--build-path" ) save_next_arg=3;;
+                "-h" | "--help" ) usage;;
+                * ) usage;;
+            esac
+        fi
+    done
+}
+
+process_args "$@"
+
+[[ -z "$PACKAGE_OS" ]] && { print_error 'OS is a required parameter'; exit 1; }
+[[ -z "$PACKAGE_ARCH" ]] && { print_error 'Arch is a required parameter'; exit 1; }
+[[ -z "$BUILD_PATH" ]] && { print_error 'Build path is a required parameter'; exit 1; }
+
+set -e
+
+# Get directory of running script
+DIR="$(cd "$(dirname "$0")" && pwd)"
+
+BUILD_REPOSITORY_LOCALPATH="$(realpath "${BUILD_REPOSITORY_LOCALPATH:-$DIR/../..}")"
+
+CMAKE_ARGS='-DCMAKE_BUILD_TYPE=Release'
+CMAKE_ARGS="$CMAKE_ARGS -DBUILD_SHARED=On -Drun_unittests=Off -Duse_default_uuid=On -Duse_emulator=Off -Duse_http=Off"
```

Comment:
You probably don't need `-DBUILD_SHARED=On` or `-Duse_default_uuid=On -Duse_emulator=Off` unless you are building the HSM library. You also don't want to set "BUILD_SHARED=On" if you are targeting MUSL.

Reply:
Wait so these CMAKE_ARGS can be removed altogether. This build script doesn't need to use cmake for the broker. I should have removed it.